extension/src/plots/model/collect.ts

```diff
@@ -216,6 +219,94 @@ export const collectData = (output: PlotsOutput): DataAccumulator => {
   return acc
 }
 
+type ComparisonPlotsAcc = {
+  [pathLabel: string]: { path: string; revisions: ComparisonRevisionData }
+}
+
+const collectSelectedComparisonPlot = ({
+  acc,
+  id,
+  isMulti,
+  path,
+  pathLabel,
+  getComparisonPlotImg
+}: {
+  acc: ComparisonPlotsAcc
+  id: string
+  isMulti: boolean
+  path: string
+  pathLabel: string
+  getComparisonPlotImg: (id: string, path: string) => ComparisonPlotImg
+}) => {
+  const doesPlotExist = acc[pathLabel].revisions[id]
+  if (!doesPlotExist) {
+    acc[pathLabel].revisions[id] = {
+      id,
+      imgOrImgs: [],
+      isMulti
```

Comment:
[Q] could the data format be `{ id, imgs }`? do we really need to differentiate between the two and could we always save under a directory? In order to differentiate in the webview we would just need to check the length of `imgs `.

Reply:
I can't spot where `isMulti` is used in the webview.

Reply:
> > could we always save under a directory

> Oh, yes that makes sense. I think grouping images in comparisonData will make handling the path data easier as well.

After trying this, I realized that this can't work correctly because we need to send missing path plots to the frontend. So our image grouping can only happen when we're collecting selected plots since that's when we know the list of paths.